src/providerhub/azext_providerhub/generated/_params.py

```diff
@@ -0,0 +1,347 @@
+        c.argument('provider_namespace', type=str, help='The name of the resource provider hosted within ProviderHub.',
+                   id_part='name')
+
+    with self.argument_context('providerhub resource-type-registration list') as c:
+        c.argument('provider_namespace', type=str,
+                   help='The name of the resource provider hosted within ProviderHub.')
+
+    with self.argument_context('providerhub resource-type-registration show') as c:
+        c.argument('provider_namespace', type=str, help='The name of the resource provider hosted within ProviderHub.',
+                   id_part='name')
+        c.argument('resource_type', type=str, help='The resource type.', id_part='child_name_1')
+
+    with self.argument_context('providerhub resource-type-registration create') as c:
+        c.argument('provider_namespace', type=str,
+                   help='The name of the resource provider hosted within ProviderHub.')
+        c.argument('resource_type', type=str, help='The resource type.')
+        c.argument('routing_type', arg_type=get_enum_type(['Default', 'ProxyOnly', 'HostBased', 'Extension',
+                                                           'Tenant', 'Fanout', 'LocationBased', 'Failover',
+                                                           'CascadeExtension']), help='')
+        c.argument('regionality', arg_type=get_enum_type(['NotSpecified', 'Global', 'Regional']), help='')
+        c.argument('endpoints', action=AddResourceTypeEndpointProperties, nargs='+', help='The resource '
+                   'type endpoint properties.')
+        c.argument('marketplace_type', arg_type=get_enum_type(['NotSpecified', 'AddOn', 'Bypass', 'Store']), help='')
+        c.argument('swagger_specifications', action=AddSwaggerSpecifications, nargs='+', help='')
+        c.argument('allowed_unauthorized_actions', nargs='+', help='')
+        c.argument('authorization_action_mappings', action=AddAuthorizationActionMappings, nargs='+', help='')
+        c.argument('linked_access_checks', action=AddLinkedAccessChecks, nargs='+', help='')
+        c.argument('default_api_version', type=str, help='')
+        c.argument('logging_rules', type=validate_file_or_dict, help=' Expected value: json-string/@json-file.')
+        c.argument('throttling_rules', action=AddThrottlingRules, nargs='+', help='')
+        c.argument('required_features', nargs='+', help='')
+        c.argument('enable_async_operation', arg_type=get_three_state_flag(), help='')
+        c.argument('provisioning_state', arg_type=get_enum_type(['NotSpecified', 'Accepted', 'Running', 'Creating',
+                                                                 'Created', 'Deleting', 'Deleted', 'Canceled',
+                                                                 'Failed', 'Succeeded', 'MovingResources',
+                                                                 'TransientFailure', 'RolloutInProgress']), help='')
```

Comment:
`provisioning_state` should be a readonly property and should be removed in create command. Please also check your swagger file and you may need to fix it as well.